### ISSUE TITLE
Adding extension to derivative filename

### ISSUE THREAD

**Issue by cybertoast** [Ssawa/imagetask#1]
@Ssawa - is there a simple way to allow the file-format/extension to be appended to the filename/url? I see I can override the key() and create_from_serial_data() methods to add the extension like:

```
    @property
    def key(self):
        return self.serialize() + self.save_options.get("format")
```
(an the associated decoding on the create_from_serial_data), but I wonder if there's a more idiomatic way you were envisioning this to be implemented.

To be clear, what I'm looking to do is return /imagetask/.<itsdangerous-serialized-id>.jpg, rather than just the serialized value without an extension, and also save the filename this way.

Thanks!

**Comment by cj-dimaggio** [comment]
Yeah I think that's probably the best way of doing it right now. I would probably do something like the following:

```python
class SpecWithExtension(ImageSpec):
    def deduceExtension(self):
        default_extension = self.image_path.split('.')[-1]
        return self.save_options.get("format", default_extension)

    @property
    def key(self):
        return '%s.%s' % (super(SpecWithExtension, self).key, self.deduceExtension())

    @classmethod
    def create_from_serial_data(cls, app, serial_data):
        stripped = '.'.join(serial_data.split('.')[:-1])
        return super(SpecWithExtension, cls).create_from_serial_data(app, stripped)
```

I think that should do the job and still work with all of the existing handlers. it's a little finicky and hacky which is why I was avoiding putting in something like it into the main Spec class but maybe it's worth just throwing in if it's useful.

**Comment by cybertoast** [comment]
Very close to what I implemented. I think it's worth having this as an
example for those who need extensions to their filenames, but certainly not
necessary as default implementation. Thanks!
